horovod/torch/mpi_ops.py

```diff
@@ -62,6 +65,12 @@ def _allreduce_function_factory(tensor):
 
 
 def _allreduce_async(tensor, output, average, name):
+    if tensor.dtype == torch.float16 and \
+            LooseVersion(torch.__version__) < LooseVersion('0.4.2'):
```

Comment:
Is this parsing fast enough to do 1000 times per second?

Reply:
I'll cache it on import.